### ISSUE TITLE
Compression Support

### ISSUE THREAD

**Issue by daplho** [logstash-plugins/logstash-output-http#63]
I'd like to propose gzip compression support for the plugin. More specifically, a configuration option like `compression => true` could enable http gzip compression for any message's payload.

**Comment by daplho** [comment]
An possible approach to introduce a config option:

```
compression => true
```

that will execute the following code in `http.rb` just before the request is sent off:

```
# Compress the body and add appropriate header
if @custom_compression == true
  headers["Content-Encoding"] = "gzip"
  httpBody = gzip(body)
else
  httpBody = body
end
```

where the `gzip` method does simply:

```
def gzip(data)
  gz = StringIO.new
  z = Zlib::GzipWriter.new(gz)
  z.write(data)
  z.close
  gz.string
end
```

Unfortunately, Manticore (and the underlying CloseableHttpClient) are not playing nice and are creating an invalid gzip string. Anybody here have some expertise in the area? I've tried lots of combinations of the options that `http_client_builder` and `request_builder` expose without any success so far.

**Comment by markwalkom** [comment]
Are you willing to make a PR with what you have?
Then we can get someone to look at this and make some suggestions to see if we can resolve and merge :)

**Comment by daplho** [comment]
Yes, here is a link:

https://github.com/logstash-plugins/logstash-output-http/pull/66

**Comment by daplho** [comment]
@markwalkom does that code look like it might be something to start with?

**Comment by ph** [comment]
Fixed by #72 and #66, thanks @daplho for the work!

**Comment by daplho** [comment]
You're welcome. Thank you as well for your contributions especially in the testing suite.
